### PR TITLE
Introducing Ngrx init state | Jira ticket: RIUI-1722

### PR DESCRIPTION
**Pull request by frontr-uk**
Following guidelines: https://ngrx.io/guide/store/install

 `npm install @ngrx/store --save`

`ng add @ngrx/store`
```
Skipping installation: Package already installed
CREATE src/app/reducers/index.ts (359 bytes)
UPDATE src/app/app.module.ts (480 bytes)
UPDATE package.json (1335 bytes)
```
`ng add @ngrx/effects`
```
+ @ngrx/effects@7.4.0
Installed packages for tooling via npm.
CREATE src/app/app.effects.spec.ts (571 bytes)
CREATE src/app/app.effects.ts (180 bytes)
UPDATE src/app/app.module.ts (613 bytes)
UPDATE package.json (1366 bytes)
```
`npm install @ngrx/store @ngrx/effects @ngrx/store-devtools @ngrx/router-store --save`

```
+ @ngrx/store@7.4.0
+ @ngrx/effects@7.4.0
+ @ngrx/router-store@7.4.0
+ @ngrx/store-devtools@7.4.0
added 2 packages from 1 contributor, updated 2 packages and audited 42616 packages in 17.396s
```

`run `npm audit fix` to fix them, or `npm audit` for details`

Creating two commits. Once that shows what tooling does. The other manually moving folders to follow  PUI manager app structure by having everything under `/store` folder in the `app` folder

**Folder structure:** 

```
src
-app
--components
--containers
--app.module.ts
--app.routes.ts
--store

-featureA
--components
--containers
--directives
--selectors
--store
--feature.module.ts
--feature.route.ts
```
Here is a link to the PUI manager app structure: https://github.com/hmcts/prd-pui-manager/tree/master/src/app

### CODE REVIEW

**Comment by matthewspeake-hmcts** [comment]
@frontr-uk great that you created via the commandline and detailed it, but we don't have the folder structure which we want out of this ticket.

**Comment by frontr-uk** [comment]
> Please add the required folder structure too.

I had moved the files into the structure that we agreed as I understood the core state inside /app/store ?
I also mentioned on the PR description:

"Creating two commits. Once that shows what tooling does. The other manually moving folders to follow PUI manager app structure by having everything under /store folder in the app folder"

**Comment by frontr-uk** [comment]
> I think we should provide router state by default with all the actions...

We are following agile and what is asked on Jira. So this was out of scope.

File: src/app/services/configuration.services.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  setConfiguration() {
+    this.store.pipe(select(fromApp.getAppFeatures), take(1)).subscribe(config => {
+      this.configuration = config;
+    });
+  }
+  getFeatureToggle() {
+    return this.configuration.features;
+  }
+
+  // todo make it global and make it make sense
```
Comment (frontr-uk) [inline src/app/services/configuration.services.ts]: I think it is best to stick the // todo in Jira so you can keep track

Comment (fazanki) [reply src/app/services/configuration.services.ts]: 👍

**Pull request merged**